shared/uavobjectdefinition/hwshared.xml

```diff
@@ -0,0 +1,41 @@
+<xml>
+	<object name="HwShared" singleinstance="true" settings="true" category="HardwareSettings">
+		<description>Templates for common enums.</description>
+
+		<field name="PortTypes" units="function" type="enum" elements="1"  defaultvalue="Disabled">
+			<options>
+				<option>Disabled</option>
+				<option>Telemetry</option>
+				<option>GPS</option>
+				<option>DSM</option>
+				<option>S.Bus</option>
+				<option>DebugConsole</option>
+				<option>ComBridge</option>
+				<option>MavLinkTX</option>
+				<option>MavLinkTX_GPS_RX</option>
+				<option>HoTT SUMD</option>
+				<option>HoTT SUMH</option>
+				<option>HoTT Telemetry</option>
+				<option>FrSKY Sensor Hub</option>
+				<option>FrSKY SPort Telemetry</option>
+				<option>LighttelemetryTx</option>
+				<option>PicoC</option>
+<!-- These last couple are special cases, really.  le sigh -->
+				<option>I2C</option>
```

Comment:
for this the top level code will need to know whether a port is set to I2C in order to know whether to try probing for an external mag etc. that being said this isn't done currently so not a huge issue :)